etl/models/document_tuple.py

```diff
@@ -18,35 +18,28 @@ def from_records(
         cls,
         *,
         records: tuple[Record, ...],
-        record_type: RecordType,
         enrichment_type: EnrichmentType,
```

Comment:
You could make this more obvious by having a `record_content: Callable[[Record], str]` signature here.